diamonds-viewer/src/components/GameBoard.js

```diff
@@ -0,0 +1,126 @@
+import React from "react";
+import { connect } from "react-redux";
+import _ from "lodash";
+import updateBoard from "../actions/updateBoard";
+import Board from "../blocks/Board";
+import {
+  base,
+  botBaseDiamond,
+  botBase,
+  diamond,
+  diamondRed,
+  botDiamond,
+  robot,
+  teleporter,
+  wall,
+  redButton
+} from "../images";
+
+class GameBoard extends React.Component {
+  componentDidMount = () => {
+    this.interval = setInterval(this.shouldUpdateBoard, 250);
+  };
+
+  componentWillUnmount = () => {
+    clearInterval(this.interval);
+  };
+
+  shouldUpdateBoard = () => {
+    const { isFetching } = this.props.board;
+    const boardId = 1;
+
+    if (!isFetching) {
+      this.props.updateBoard(boardId);
+    }
+  };
+
+  decideCharacter = content => {
+    const goImgMap = {
+      Teleporter: teleporter,
+      Wall: wall,
+      DiamondButton: redButton
+    };
+
+    if (
+      _.has(content, "botName") &&
+      _.has(content, "base") &&
+      _.has(content, "diamond")
+    ) {
+      return botBaseDiamond;
+    } else if (_.has(content, "botName") && _.has(content, "base")) {
+      return botBase;
+    } else if (_.has(content, "botName") && _.has(content, "diamond")) {
+      return botDiamond;
+    } else if (_.has(content, "base")) {
+      return base;
+    } else if (_.has(content, "botName")) {
+      return robot;
+    } else if (_.has(content, "diamond")) {
+      return content.points === 1 ? diamond : diamondRed;
+    } else if (_.has(content, "go")) {
+      return goImgMap[content.goName];
+    } else {
+      return null;
+    }
+  };
+
+  decideCharacterName = content => {
+    if (_.has(content, "botName")) {
+      return content.botName;
+    } else if (_.has(content, "base")) {
+      return content.base;
+    } else {
+      return null;
+    }
+  };
+
+  render = () => {
+    const { rows, width } = this.props.board;
+    const bigCellSize = (90 / width).toFixed(2);
+    const smallCellSize = (60 / width).toFixed(2);
+
+    return (
+      <Board>
+        {rows.map((cells, key) => {
+          return (
+            <Board.Row key={key}>
```

Comment:
Can we use a [CSS grid](https://developer.mozilla.org/en-US/docs/Web/CSS/grid) here rather than emulate tables? That way we could keep a single mounted component for each bot's whole lifetime (=> no more requesting every asset for each re-render, would help with adding transition animations).

Reply:
They dont rerender. Only the cells that change will rerender. You can inspect it and see how that works.

Reply:
Also, regarding transition animations, to make that work smoothly every move done would be needed to known by the client. So I'm thinking that instead of the client sending a request every 250ms we could move to server-sent events on every event that changes the board. We will then remove a lot of overhead, and we could show every single move done. But that is also way out of scope for this pull request.

Reply:
> Only the cells that change will rerender.

Sorry, I wasn't clear here. The problem is that bots (and other game objects) are unmounted and remounted each time they move, breaking everything that relies on a consistent identity (such as animations and other transitions). 

Yes, moving to push-based websockets would also improve this, but the current janky movement becomes even more confusing due to the lack of continuity between visual states.

Reply:
I think its janky in the same way Tetris is janky. The problem with understanding what happens comes more from that you never see all moves, not that the transitions are smooth. If changes were made so that the viewer were notified of every move the confusion would most likely disappear.  Other easier solutions could also be made to make it easier to follow what happens, such as coloring robots and bases etc. Which are more trivial solutions then making so that things arn't mounted/unmounted.

If you want smooth transitions without having the modify the DOM-tree as soon as bot moves you would probably have to move to some WebGL or Canvas renderer, which is not a trivial solution by any means. But, still possible of course.